homework_12/homework_12.py

```diff
@@ -0,0 +1,167 @@
+# 1. Використати файл як базу даних для збереження записів телефонної книги із попередніх завдань.
+#
+# Ваша телефонна книга, що до цього містилася в dict, має зберігатися у вигляді тексту в JSON форматі.
+# При закритті програми і повторному відкритті всі попередні дані мають бути доступними.
+# Підказка: Ви можете використати бібліотеку json, яка допоможе із перетворенням dict в JSON string (при записі в файл)
+# і JSON string в dict (при читанні із файлу). Файл слід оновлювати після кожної успішної операції add або delete.
+
+import json
+import time
+
+
+
+def ubd_dict(data_j):
+    data_j = json.dumps(data_j)
+    with open('phone_book.json', 'w+') as new_file:
+        new_file.write(data_j)
+    return
+
+
+def r_dict():
+    with open('phone_book.json', 'r+') as file:
+        text = file.read()
+        if text == '':
+            return {}
+
+        else:
+            json_data = json.loads(text)
+            return json_data
+
+
+while True:
+    user_input = input('\n Select and enter a command \n'
+                       '(stats, add, delete, list, show): ')
+    print(user_input)
+
+    try:
+        file = open('phone_book.json')
```

Comment:
You never close this file description. I would suggest using the open function as a context manager here. Also, you can move this part outside the loop, you don't need this action every iteration